pkg/util/normalize.go

```diff
@@ -4,32 +4,52 @@ import (
 	"fmt"
 	"net"
 	"net/url"
+	"strings"
 )
 
 // NormalizeAppURL normalizes localhost and 127.0.0.1 as the same string.  This
 // ensures that we don't add duplicate apps.
-func NormalizeAppURL(u string) string {
+func NormalizeAppURL(u string, forceHTTPS bool) string {
 	parsed, err := url.Parse(u)
 	if err != nil {
 		return u
 	}
 
-	host, port, err := net.SplitHostPort(parsed.Host)
-	if err != nil {
-		return u
-	}
+	parsed = stripDeployID(*parsed)
 
-	// this shouldn't be valid: https://api.example.com:80/api/inngest
-	if parsed.Scheme == "https" && port != "" {
-		parsed.Host = host
-		return parsed.String()
+	if forceHTTPS {
+		if parsed.Scheme != "https" {
+			parsed.Scheme = "https"
+		}
 	}
 
-	switch host {
-	case "localhost", "127.0.0.1", "0.0.0.0":
-		parsed.Host = fmt.Sprintf("localhost:%s", port)
-		return parsed.String()
-	default:
-		return u
+	if strings.Contains(parsed.Host, ":") {
+		host, port, err := net.SplitHostPort(parsed.Host)
+		if err != nil {
+			return u
+		}
+
+		// this shouldn't be valid: https://api.example.com:80/api/inngest
+		if parsed.Scheme == "https" && port != "" {
+			parsed.Host = host
+			return parsed.String()
+		}
```

Comment:
It's unconventional, but it's valid to have HTTPS running on a custom port. Maybe we only strip the port here if `port == "443"`?

Reply:
Sounds reasonable to me but I'll let @tonyhb weigh in since he wrote that code